includes/admin/plugins.php

```diff
@@ -129,108 +128,52 @@ function perflab_render_plugin_card( array $plugin_data ) {
-					$action_links[] = sprintf(
-						'<a class="install-now button" data-slug="%s" href="%s" aria-label="%s" data-name="%s" data-plugin-activation-nonce="%s">%s</a>',
-						esc_attr( $plugin_data['slug'] ),
-						esc_url( $status['url'] ),
-						/* translators: %s: Plugin name and version. */
-						esc_attr( sprintf( _x( 'Install %s now', 'plugin', 'default' ), $name ) ),
-						esc_attr( $name ),
-						esc_attr( wp_create_nonce( 'perflab_activate_plugin_' . $plugin_data['slug'] ) ),
-						esc_html__( 'Install Now', 'default' )
-					);
-				} else {
-					$action_links[] = sprintf(
-						'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
-						esc_html( _x( 'Cannot Install', 'plugin', 'default' ) )
-					);
-				}
-			}
-			break;
-
-		case 'update_available':
-		case 'latest_installed':
-		case 'newer_installed':
-			if ( is_plugin_active( $status['file'] ) ) {
-				$action_links[] = sprintf(
-					'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
-					esc_html( _x( 'Active', 'plugin', 'default' ) )
-				);
-				if ( current_user_can( 'deactivate_plugin', $status['file'] ) ) {
-					$s       = isset( $_REQUEST['s'] ) ? $_REQUEST['s'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-					$context = $status['status'];
-
-					$action_links[] = sprintf(
-						'<a href="%s" id="deactivate-%s" aria-label="%s">%s</a>',
-						esc_url(
-							add_query_arg(
-								array(
-									'_wpnonce' => wp_create_nonce( 'perflab_deactivate_plugin_' . $status['file'] ),
-									'action'   => 'perflab_deactivate_plugin',
-									'plugin'   => $status['file'],
-								),
-								network_admin_url( 'plugins.php' )
-							)
-						),
-						esc_attr( $plugin_data['slug'] ),
-						/* translators: %s: Plugin name. */
-						esc_attr( sprintf( _x( 'Deactivate %s', 'plugin', 'default' ), $plugin_data['slug'] ) ),
-						esc_html__( 'Deactivate', 'default' )
-					);
-				}
-			} elseif ( current_user_can( 'activate_plugin', $status['file'] ) ) {
-				if ( $compatible_php && $compatible_wp ) {
-					$button_text = __( 'Activate', 'default' );
-					/* translators: %s: Plugin name. */
-					$button_label = _x( 'Activate %s', 'plugin', 'default' );
-					$activate_url = add_query_arg(
-						array(
-							'_wpnonce' => wp_create_nonce( 'perflab_activate_plugin_' . $status['file'] ),
-							'action'   => 'perflab_activate_plugin',
-							'plugin'   => $status['file'],
-						),
-						network_admin_url( 'plugins.php' )
-					);
+	if ( is_plugin_active( $status['file'] ) ) {
+		$action_links[] = sprintf(
+			'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+			esc_html( _x( 'Active', 'plugin', 'default' ) )
+		);
+	} elseif (
+		$compatible_php &&
+		$compatible_wp &&
+		current_user_can( 'activate_plugin', $status['file'] ) &&
```

Comment:
Thinking about this further, we need to be careful here. We can only check for the specific `activate_plugin` capability if the plugin is already installed. If it's not installed, `$status['file']` will be empty. In that case we need to fall back to the generic `activate_plugins` capability.
```suggestion
		(
			$status['file'] && current_user_can( 'activate_plugin', $status['file'] ) ||
			current_user_can( 'activate_plugins' )
		) &&
```